source/Assets/Plugins/Didomi/IOS/Didomi.mm

```diff
@@ -20,6 +14,72 @@
      return res;
 }
 
+/**
+ Method used to create a dictionary based on an instance of DDMUserStatusIDs.
+ */
+NSDictionary<NSString *, NSArray<NSString *> *> * CreateDictionaryFromStatusIDs(DDMUserStatusIDs *ids)
+{
+    return @{
+        @"enabled": [[ids enabled] allObjects],
+        @"disabled": [[ids disabled] allObjects]
+    };
+}
+
+/**
+ Method used to create a string from a dictionary that doesn't have the form <NSString *, NSString *>.
+ */
+char* ConvertComplexDictionaryToString(NSDictionary * dataDict)
+{
+   NSData *jsonData = [NSJSONSerialization dataWithJSONObject:dataDict options:NSJSONWritingPrettyPrinted error:nil];
+   NSString *jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+   NSLog(@"jsonData dictionary as string:\n%@", jsonString);
+   return cStringCopy([jsonString UTF8String]);
+}
+
+/**
+ Method used to map user status from DDMUserStatus to a JSON string.
+ */
+char* MapUserStatus(DDMUserStatus *userStatus) {
+    // String properties
+    NSString *userID = [userStatus userID];
+    NSString *created = [userStatus created];
+    NSString *updated = [userStatus updated];
+    NSString *consentString = [userStatus consentString];
+    NSString *additionalConsent = [userStatus additionalConsent];
+    
+    DDMUserStatusPurposes *purposeStatus = [userStatus purposes];
+    DDMUserStatusVendors *vendorsStatus = [userStatus vendors];
+    
+    NSDictionary *dictionary = @{
+        @"user_id": userID,
+        @"created" : created,
+        @"updated": updated,
+        @"consent_string": consentString,
+        @"addtl_consent": additionalConsent,
+        @"purposes": @{
+            @"consent": CreateDictionaryFromStatusIDs([purposeStatus consent]),
+            @"legitimate_interest": CreateDictionaryFromStatusIDs([purposeStatus legitimateInterest]),
+            @"global": CreateDictionaryFromStatusIDs([purposeStatus global]),
+            @"essential": [[purposeStatus essential] allObjects]
+        },
+        @"vendors": @{
+            @"consent": CreateDictionaryFromStatusIDs([vendorsStatus consent]),
+            @"legitimate_interest": CreateDictionaryFromStatusIDs([vendorsStatus legitimateInterest]),
+            @"global": CreateDictionaryFromStatusIDs([vendorsStatus global]),
+            @"global_consent": CreateDictionaryFromStatusIDs([vendorsStatus globalConsent]),
+            @"global_li": CreateDictionaryFromStatusIDs([vendorsStatus globalLegitimateInterest])
+        }
+    };
+    
+    return ConvertComplexDictionaryToString(dictionary);
+}
+
+/**
```

Comment:
@felipe-saez  Was this comment moved on purpose?

Reply:
@pmerlet-at-didomi initially it was above cStringCopy which I thought didn't make much sense. From what I understood the comment applies to the methods within `extern "C" {`. That's why I moved it.